packages/foam-vscode/src/features/commit-on-save.ts

```diff
@@ -0,0 +1,100 @@
+import { commands, ExtensionContext, TextDocument, workspace, extensions } from "vscode";
+import { GitExtension, Commit } from "../git";
+import { FoamFeature } from "../types";
+import { exec } from 'child_process';
+import { promisify } from 'util';
+
+const execP = promisify(exec);
+let latestCommitTime;
+let latestSyncTime;
+let commitInterval = 10; // replace with configuration
+let syncInterval = 60 * 5;
+
+const feature: FoamFeature = {
+  activate: (context: ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand("foam-vscode.run-hook-on-save", commitOnSave)
+    );
+    workspace.onDidSaveTextDocument((document: TextDocument) => {
+      commands.executeCommand("foam-vscode.run-hook-on-save", document);
+    });
+  }
+};
+
+async function commitOnSave (document: TextDocument) {
+  let amend = false;
+
+  const gitExtension = extensions.getExtension<GitExtension>('vscode.git').exports;
+  const git = gitExtension.getAPI(1);
+  // get current repo instance
+  const repo = git.repositories[0];
```

Comment:
a few questions about some edge cases:
- if we are not in a git repo, I imagine this would fail?
- when would this array have more than one element? 
- what happens if someone has multiple git repos as subdirectories (something similar to what @scott-joe suggested somewhere)? Scott, can you share here your use case for reference?
- as we currently support workspaces, it's possible/likely they will live in separate directories/repos

Sorry about all the nitpicking, my fear is that if we add this feature we could end up with a bunch of edge cases, and we need to be clear about what we support/bless and what we don't (and of course we shouldn't break in any case :) )

Reply:
> if we are not in a git repo, I imagine this would fail?

Probably we should throw an error and suggest to create repository.

> as we currently support workspaces, it's possible/likely they will live in separate directories/repos

Would be great to have a reference, to test it.

> when would this array have more than one element?

It will pick first repo, that is detected. [Here's source code](https://github.com/microsoft/vscode/blob/2b7bf327b9c59ef56ce7608acab637800714a039/extensions/git/src/api/api1.ts#L248).